bundles/org.openhab.binding.mail/src/main/resources/OH-INF/addon/addon.xml

```diff
@@ -6,5 +6,6 @@
 	<type>binding</type>
 	<name>Mail Binding</name>
 	<description>This binding is used to access POP3, IMAP and SMTP servers.</description>
+	<connection>cloud</connection>
```

Comment:
I would argue about this - see also https://github.com/openhab/openhab-addons/pull/14813#discussion_r1167470570

The clear "cloud" case is when using an API provided by some specific vendor, i.e. "someone else's computer". Unfortunately that leaves some ambiguity for the cases where a standard or open protocol is implemented. For remoteopenhab we decided to use the criteria "the most common use-case" and defined it as `local`.

I run my own IMAP and SMTP servers, so for me it would be `local`. It's certainly possible to use it for remote servers, but so is anything running locally, really.

So difficult to say what is the most common use-case here, but the protocols are defined in RFC's and anyone can run such servers locally - and there is no vendor lock-in in play here like for most cloud bindings. So I'm leaning towards `local`.

Reply:
With that in mind `icalendar` (also in this PR) is also `local`. Is there a connection `network`?

Reply:
The descriptions [XSD](https://www.openhab.org/schemas/addon-1.0.0.xsd) defines also `hybrid` which may also match to this use case. For Mail: in the usual case there is an internet connection required to interact with the world, but the connection itself to the mail server may be also internal. This is similar to use case of `icalendar`, there it's just dependent where the calendar is located (defined by the user) but if it's in the internet, `cloud` may also match. Different from mail is, that the calendar usually gets any interaction by other users than the owner (which may be an external person).

Reply:
No, in this case it's a choice between `local` and `cloud`:
https://github.com/jlaur/openhab-docs/blob/6cf713c1cae012e0d417c46bf2582bf6c51c7a5c/developers/addons/addon.md#xml-structure-for-add-on-definitions

I realize there is still a gap, but I would opt for interpreting `local` as "non-cloud" even if it may not always be "local".

Reply:
I'm confused, your link also describes `hybrid`.

Reply:
Yes, and `none`, but I don't think any of them would apply here. The intention of `hybrid` is not to describe ambiguity, but rather a binding that works perfectly well within your own infrastructure, but can optionally use external cloud services for extended functionality. It will still work if the cloud services are shut down.

Reply:
> I would argue about this - see also [#14813 (comment)](https://github.com/openhab/openhab-addons/pull/14813#discussion_r1167470570)
> 
> The clear "cloud" case is when using an API provided by some specific vendor, i.e. "someone else's computer". Unfortunately that leaves some ambiguity for the cases where a standard or open protocol is implemented. For remoteopenhab we decided to use the criteria "the most common use-case" and defined it as `local`.

I was thinking of this comment: https://github.com/openhab/openhab-addons/pull/14690#issuecomment-1485551835

For NTP it is also a an open/RFC defined service.... But maybe the lack of a common use case of running an NTP server locally pushed that one to the 'cloud' category?

Reply:
Why? You could also set your home router as NTP peer as these devices usually offer one, which is quite good and has stable latencies. At least i would do. Another use case would be to monitor time differences of multiple devices. No internet needed at all, if the devices implement not only sntp. Same issue as for mail or icalendar.

Reply:
@mlobstein - ouch, you beat me with my own argument. 😉 I admit than even now the documentation for `local` is unclear. I think we are mixing things:
- Location: "local" (e.g. LAN) vs. "remote" (internet).
- Ownership: my server or someone else's server.
- Protocol: open protocol defined by RFC or vendor-specific protocol.
- Maybe even more?

For me - now - "cloud" is someone else's computer not using an open protocol. You cannot run it locally, and if the service shuts down, you cannot just move to another company implementing the same service. Essentially this means that cloud bindings can die overnight. By this definition, "local" is the opposite, so either something running entirely locally, or possibly outside your LAN for whichever reason. That could include remoteopenhab in the summerhouse, a friend's FTP server or your current ISP's SMTP server.

This means that I would now back paddle on the NTP comment and redefine it as "local".

This is just my opinion, and I still think the definition/documentation is not very clear. But I think the intention was to differentiate between traditional cloud (outside your control) and something can at least could potentially run locally.

Reply:
I just merged the PR as it looks good to me, and I see no reason to block it. We can still continue this discussion and change declarations where appropriate. Perhaps it would be better to move the discussion to #14885, WDYT?

Reply:
I agree, for the remainder of the bindings, I will judge cloud to mean connecting to a proprietary remote api (either via login credentials or a read-only scraper).